In [1]:
import pandas as pd
import numpy as np
from sklearn import linear_model
from sklearn.metrics import r2_score
from sklearn.metrics import mean_squared_error
from sklearn.model_selection import cross_val_score

In [2]:
final = pd.read_csv("/Users/mdbarshay/Desktop/401/Project 3/final.csv", index_col = 0)

In [3]:
X = final.iloc[:, :50]
y = final.iloc[:,50]

In [5]:
X = (X-X.mean())/X.std()
y = y / 100000

In [21]:
lm = linear_model.LinearRegression()
res = cross_val_score(lm, X, y, scoring = "neg_mean_squared_error", cv = 5)
res2 = cross_val_score(lm, X, y, scoring = "neg_root_mean_squared_error", cv = 5)
res3 = cross_val_score(lm, X, y, scoring = "r2", cv = 5)

The average mean squared error for 5-fold cross validated linear regression is:

In [22]:
-np.mean(res)

2817.7455042857882

The average root mean squared error for 5-fold cross validated linear regression is:

In [23]:
-np.mean(res2)

48.17603766774928

The average R^2 for 5-fold cross validated linear regression is:

In [26]:
-np.mean(res3)

0.33871622420764114

This means that on average our predictions for median house price are roughly 4,800,000 dollars away from the true house price on average. Furthermore, our linear regression model explains around 33.87% of the variance in house prices. 

We are only using this model for predictions and not interpreting the coefficients and so checking the model assumptions is not neccesary.

In [27]:
final

num_room  full_sq  sport_count_5000  sport_count_3000  trc_count_5000  \
0      1.962264       43                52                21              52   
1      1.925926       34                66                19              40   
2      1.858586       43                67                20              35   
3      2.132275       89                26                18              22   
4      2.875000       77               195                77             114   
...         ...      ...               ...               ...             ...   
30287  2.000000       44                84                29              53   
30288  4.000000       86               171                80              90   
30289  1.000000       45                11                 6               4   
30290  2.000000       64                65                33              45   
30291  2.000000       43                54                26              22   

       zd_vokzaly_avto_km  sadovoe_km  sport_count_2000  kremlin_km  \
0               14.231961   13.100618                10   15.156211   
1                9.242586    6.444333                11    8.698054   
2                9.540544    6.963403                 8    9.067885   
3               17.478380   17.457198                13   19.487005   
4                1.595898    0.046810                21    2.578671   
...                   ...         ...               ...         ...   
30287           11.396447   10.543724                15   12.599292   
30288            1.010437    1.203215                36    3.269284   
30289           25.699461   19.868997                 3   21.905792   
30290           15.303338   11.752036                18   13.622569   
30291           12.243439    9.675169                12   11.812614   

       bulvar_ring_km  ...  exhibition_km  big_church_count_5000  \
0           13.675657  ...       7.023705                     13   
1            8.132640  ...       2.358840                     15   
2            8.054252  ...       4.958214                     11   
3           18.309433  ...       5.029696                      4   
4            0.787593  ...       1.339652                    135   
...               ...  ...            ...                    ...   
30287       11.118577  ...       4.877876                     15   
30288        1.874868  ...       3.543784                     98   
30289       21.038561  ...       4.555385                      2   
30290       12.872535  ...       1.800186                      6   
30291       10.228634  ...       2.477068                      7   

       sport_count_1000  office_sqm_1500  market_count_5000  \
0                     6            39554                  4   
1                     2           102910                 14   
2                     5                0                 10   
3                     3            11000                  3   
4                     7          1195735                 14   
...                 ...              ...                ...   
30287                 3           217042                  6   
30288                 8          1815698                 15   
30289                 0                0                  1   
30290                10            37800                  7   
30291                 3                0                 10   

       school_education_centers_raion  museum_km  healthcare_centers_raion  \
0                                   5   7.389498                         1   
1                                   8   0.709260                         1   
2                                   7   3.156423                         1   
3                                  10  16.041521                         1   
4                                   9   1.800125                         4   
...                               ...        ...                       ...   
30287                               4   4.836787                         